clients/pkg/promtail/utils/remotewrite_server.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"math"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/util"
+)
+
+// RemoteWriteRequest wraps the received logs remote write request that is received.
+type RemoteWriteRequest struct {
+	TenantID string
+	Request  logproto.PushRequest
+}
+
+// NewRemoteWriteServer creates and starts a new httpserver.Server that can handle remote write request. When a request is handled,
+// the received entries are written to receivedChan, and status is responded.
+func NewRemoteWriteServer(receivedChan chan RemoteWriteRequest, status int) *httptest.Server {
```

Comment:
if we're not using this anywhere else currently can we move it into a test file and name it more obviously that it's only used in tests?